autodocsumm/__init__.py

```diff
@@ -303,8 +310,14 @@ def add_autosummary(self):
                 indent = '    '
 
                 for (documenter, _) in documenters:
-                    self.add_line(
-                        indent + '~' + documenter.fullname, sourcename)
+                    if relative_ref_paths:
+                        obj_ref_path = documenter.fullname.lstrip(
+                            self.modname + '.')
```

Comment:
Ok, `documenter.fullname.lstrip(self.modname + '.')` is causing the problem by the way how `lstrip` works. it replaces any of the characters you give until it does not find them anymore. If you give here `"dummy_title."`  as argument, then `dummy_title.test_func` ends up with `st_func`. One possibility to fix this would be to exclude the `.` in the `lstrip` call:

```
                        obj_ref_path = documenter.fullname.lstrip(
                            self.modname)
                        if obj_ref_path.startswith("."):
                            obj_ref_path = obj_ref_path[1:]
```

But is your objective to just replace the `modname` in the left part? Then we should maybe avoid the call to `lstrip` as this not how `lstrip` is supposed to be used. Maybe rather something more explicit such as

```suggestion
                        parts = documenter.fullname.split(".")
                        modparts = self.modname.split(".")
                        for modpart, part in list(zip(modparts, parts)):
                            if modpart == part:
                                parts = parts[1:]
                        obj_ref_path = ".".join(parts)
```

what do you think?

Reply:
Yes, the objective is to just remove the modname. You are right, I did not use `.lstrip` correctly there. 

Your solution should work, I suppose. But, another idea, maybe we should just replace`.lstrip` with `.removeprefix`. This should do what I intended `.lstrip` to do. It's more concise and likely faster.

https://docs.python.org/3/library/stdtypes.html#str.removeprefix

Reply:
that's right, but `removeprefix` is not available for `python<3.9`. maybe we then just use the standard thing:

```suggestion
                        modname = self.modname + "."
                        if documenter.fullname.startswith(modname):
                            obj_ref_path = documenter.fullname[len(modname):]
                        else:
                            obj_ref_path = documenter.fullname
```

Reply:
Ah, yes. I missed that. Then we just implement it ourselfs, like you just did. I'd still prefer that approach.

Reply:
alright. I'll make it a bit cleaner but let's to it that way